connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryOffsetBackingStore.java

```diff
@@ -52,11 +52,15 @@ public void configure(WorkerConfig config) {
 
     @Override
     public synchronized void start() {
+        executor = Executors.newSingleThreadExecutor();
     }
 
     @Override
     public synchronized void stop() {
-        // Nothing to do since this doesn't maintain any outstanding connections/data
+        if (executor != null) {
+            executor.shutdown();
+            executor = null;
```

Comment:
30s maybe? Timeouts like these should be fairly liberal, we really just want to bound how long we're willing to wait in the worst case, but usually it shouldn't block for any significant amount of time. (All "user" callbacks for this class are internal to Connect, and they don't do anything that should take a long time).